diff -up dovecot-1.2.rc2/dovecot-example.conf.default-settings dovecot-1.2.rc2/dovecot-example.conf
--- dovecot-1.2.rc2/dovecot-example.conf.default-settings	2009-04-06 10:45:53.519651253 +0200
+++ dovecot-1.2.rc2/dovecot-example.conf	2009-04-06 10:45:53.525647967 +0200
@@ -10,17 +10,14 @@
 # value inside quotes, eg.: key = "# char and trailing whitespace  "
 
 # Default values are shown for each setting, it's not required to uncomment
-# any of the lines. Exception to this are paths, they're just examples with
-# the real defaults being based on configure options. The paths listed here
-# are for configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var
-# --with-ssldir=/etc/ssl
+# any of the lines.
 
 # Base directory where to store runtime data.
 #base_dir = /var/run/dovecot/
 
 # Protocols we want to be serving: imap imaps pop3 pop3s managesieve
 # If you only want to use dovecot-auth, you can set this to "none".
-#protocols = imap imaps
+#protocols = imap imaps pop3 pop3s
 
 # A space separated list of IP or host addresses where to listen in for
 # connections. "*" listens in all IPv4 interfaces. "[::]" listens in all IPv6
@@ -42,13 +39,13 @@
 #     listen = *:12000
 #     ..
 #   }
-#listen = *
+#listen = *, [::]
 
 # Disable LOGIN command and all other plaintext authentications unless
 # SSL/TLS is used (LOGINDISABLED capability). Note that if the remote IP
 # matches the local IP (ie. you're connecting from the same computer), the
 # connection is considered secure and plaintext authentication is allowed.
-#disable_plaintext_auth = yes
+#disable_plaintext_auth = no
 
 # Should all IMAP and POP3 processes be killed when Dovecot master process
 # shuts down. Setting this to "no" means that Dovecot can be upgraded without
@@ -95,8 +92,8 @@
 # dropping root privileges, so keep the key file unreadable by anyone but
 # root. Included doc/mkcert.sh can be used to easily generate self-signed
 # certificate, just make sure to update the domains in dovecot-openssl.cnf
-#ssl_cert_file = /etc/ssl/certs/dovecot.pem
-#ssl_key_file = /etc/ssl/private/dovecot.pem
+#ssl_cert_file = /etc/pki/dovecot/certs/dovecot.pem
+#ssl_key_file = /etc/pki/dovecot/private/dovecot.pem
 
 # If key file is password protected, give the password here. Alternatively
 # give it when starting dovecot with -p parameter. Since this file is often
@@ -485,7 +482,7 @@
 # locking methods as well. Some operating systems don't allow using some of
 # them simultaneously.
 #mbox_read_locks = fcntl
-#mbox_write_locks = dotlock fcntl
+#mbox_write_locks = fcntl
 
 # Maximum time in seconds to wait for lock (all of them) before aborting.
 #mbox_lock_timeout = 300
diff -up dovecot-1.2.rc2/src/master/master-settings.c.default-settings dovecot-1.2.rc2/src/master/master-settings.c
--- dovecot-1.2.rc2/src/master/master-settings.c.default-settings	2009-04-06 10:45:53.523648309 +0200
+++ dovecot-1.2.rc2/src/master/master-settings.c	2009-04-06 10:48:05.630586332 +0200
@@ -178,8 +178,8 @@ struct settings default_settings = {
 	MEMBER(syslog_facility) "mail",
 
 	/* general */
-	MEMBER(protocols) "imap imaps",
-	MEMBER(listen) "*",
+	MEMBER(protocols) "imap imaps pop3 pop3s",
+	MEMBER(listen) "*, [::]",
 	MEMBER(ssl_listen) "",
 
 	MEMBER(ssl) "yes",
@@ -191,7 +191,7 @@ struct settings default_settings = {
 	MEMBER(ssl_cipher_list) "",
 	MEMBER(ssl_cert_username_field) "commonName",
 	MEMBER(ssl_verify_client_cert) FALSE,
-	MEMBER(disable_plaintext_auth) TRUE,
+	MEMBER(disable_plaintext_auth) FALSE,
 	MEMBER(verbose_ssl) FALSE,
 	MEMBER(shutdown_clients) TRUE,
 	MEMBER(nfs_check) TRUE,
@@ -255,7 +255,7 @@ struct settings default_settings = {
 	MEMBER(maildir_copy_preserve_filename) FALSE,
 	MEMBER(maildir_very_dirty_syncs) FALSE,
 	MEMBER(mbox_read_locks) "fcntl",
-	MEMBER(mbox_write_locks) "dotlock fcntl",
+	MEMBER(mbox_write_locks) "fcntl",
 	MEMBER(mbox_lock_timeout) 300,
 	MEMBER(mbox_dotlock_change_timeout) 120,
 	MEMBER(mbox_min_index_size) 0,
