--- ./dovecot-1.0.beta2/src/auth/passdb-pam.c.pam-tty	2006-01-30 11:02:05.000000000 +0100
+++ ./dovecot-1.0.beta2/src/auth/passdb-pam.c	2006-01-30 11:03:08.000000000 +0100
@@ -256,6 +256,10 @@
 		if (host != NULL)
 			pam_set_item(pamh, PAM_RHOST, host);
 
+		/* fix bug 149673, need dummy TTY for pam_access,
+		   FIXME: should we check and report an error for PAM_TTY? */
+		pam_set_item(pamh, PAM_TTY, "dovecot");
+
 		status = pam_auth(request, pamh, &str);
 		if ((status2 = pam_end(pamh, status)) == PAM_SUCCESS) {
 			/* FIXME: check for PASSDB_RESULT_UNKNOWN_USER
