diff -up dovecot-2.0.beta4/src/lib-sql/Makefile.am.betahotfix dovecot-2.0.beta4/src/lib-sql/Makefile.am
--- dovecot-2.0.beta4/src/lib-sql/Makefile.am.betahotfix	2010-05-05 04:00:03.000000000 +0200
+++ dovecot-2.0.beta4/src/lib-sql/Makefile.am	2010-05-05 11:34:17.286805721 +0200
@@ -38,13 +38,13 @@ if ! SQL_PLUGINS
 driver_sources = \
 	driver-mysql.c \
 	driver-pgsql.c \
-	driver-sqlpool.c \
 	driver-sqlite.c
 endif
 
 libsql_la_SOURCES = \
 	$(dist_sources) \
-	$(driver_sources)
+	$(driver_sources) \
+	driver-sqlpool.c
 nodist_libsql_la_SOURCES = sql-drivers-register.c
 
 if SQL_PLUGINS
