--- dovecot-1.1.2/dovecot-example.conf.passwd	2008-07-29 14:30:57.000000000 +0200
+++ dovecot-1.1.2/dovecot-example.conf	2008-07-29 14:32:48.000000000 +0200
@@ -92,6 +92,8 @@
 
 # If key file is password protected, give the password here. Alternatively
 # give it when starting dovecot with -p parameter.
+# Security note: if you have this config file world readable, then put the
+# password into /etc/sysconfig/dovecot.
 #ssl_key_password =
 
 # File containing trusted SSL certificate authorities. Set this only if you
