--- dovecot-1.0.beta2/configure.in.lib64	2006-03-01 15:21:21.000000000 +0100
+++ dovecot-1.0.beta2/configure.in	2006-03-01 15:23:13.000000000 +0100
@@ -1354,11 +1354,11 @@
 				PGSQL_INCLUDE=$i/$j
 			fi
 		done
-		for j in lib lib/pgsql lib/postgres lib/postgresql ""; do
+		for lib in lib64 lib; do for j in $lib $lib/pgsql $lib/postgres $lib/postgresql ""; do
 			if test -f "$i/$j/libpq.so" || test -f "$i/$j/libpq.a"; then
 				PGSQL_LIBDIR=$i/$j
 			fi
-		done
+		done; done
 	done
 
 	old_LIBS=$LIBS
@@ -1397,11 +1397,11 @@
 				MYSQL_INCLUDE=$i/$j
 			fi
 		done
-		for j in lib lib/mysql ""; do
+		for lib in lib64 lib; do for j in $lib $lib/mysql ""; do
 			if test -f "$i/$j/libmysqlclient.so" || test -f "$i/$j/libmysqlclient.a"; then
 				MYSQL_LIBDIR=$i/$j
 			fi
-		done
+		done; done
 	done
 
 	old_LIBS=$LIBS
@@ -1465,11 +1465,11 @@
 				MYSQL_INCLUDE=$i/$j
 			fi
 		done
-		for j in lib lib/sqlite3 ""; do
+		for lib in lib64 lib; do for j in $lib $lib/sqlite3 ""; do
 			if test -f "$i/$j/libsqlite3.so" || test -f "$i/$j/libsqlite3.a"; then
 				MYSQL_LIBDIR=$i/$j
 			fi
-		done
+		done; done
 	done
 
 	old_LIBS=$LIBS
