diff -up dovecot-2.0.beta4/dovecot-2-0-pigeonhole-ece958a18920/src/managesieve-login/client.c.betahotfix dovecot-2.0.beta4/dovecot-2-0-pigeonhole-ece958a18920/src/managesieve-login/client.c
--- dovecot-2.0.beta4/dovecot-2-0-pigeonhole-ece958a18920/src/managesieve-login/client.c.betahotfix	2010-04-21 18:01:20.668855900 +0200
+++ dovecot-2.0.beta4/dovecot-2-0-pigeonhole-ece958a18920/src/managesieve-login/client.c	2010-04-21 18:01:20.689708242 +0200
@@ -32,6 +32,13 @@ const char *login_protocol = "managesiev
 const char *login_process_name = "managesieve-login";
 unsigned int login_default_port = 4190;
 
+const struct login_binary login_binary = {
+	.protocol = "managesieve",
+	.process_name = "managesieve-login",
+	.default_port = 4190,
+	.default_ssl_port = 0
+};
+
 void login_process_preinit(void)
 {
 	login_set_roots = managesieve_login_settings_set_roots;
