--- dovecot-1.0.rc7/src/master/master-settings.c.default-settings	2006-08-10 21:14:55.000000000 +0200
+++ dovecot-1.0.rc7/src/master/master-settings.c	2006-09-22 11:31:47.000000000 +0200
@@ -268,8 +268,8 @@
 	MEMBER(syslog_facility) "mail",
 
 	/* general */
-	MEMBER(protocols) "imap imaps",
-	MEMBER(listen) "*",
+	MEMBER(protocols) "imap imaps pop3 pop3s",
+	MEMBER(listen) "[::]",
 	MEMBER(ssl_listen) "",
 
 	MEMBER(ssl_disable) FALSE,
@@ -280,7 +280,7 @@
 	MEMBER(ssl_parameters_regenerate) 168,
 	MEMBER(ssl_cipher_list) "",
 	MEMBER(ssl_verify_client_cert) FALSE,
-	MEMBER(disable_plaintext_auth) TRUE,
+	MEMBER(disable_plaintext_auth) FALSE,
 	MEMBER(verbose_ssl) FALSE,
 	MEMBER(shutdown_clients) TRUE,
 	MEMBER(nfs_check) TRUE,
@@ -335,7 +335,7 @@
 	MEMBER(maildir_stat_dirs) FALSE,
 	MEMBER(maildir_copy_with_hardlinks) FALSE,
 	MEMBER(mbox_read_locks) "fcntl",
-	MEMBER(mbox_write_locks) "dotlock fcntl",
+	MEMBER(mbox_write_locks) "fcntl",
 	MEMBER(mbox_lock_timeout) 300,
 	MEMBER(mbox_dotlock_change_timeout) 120,
 	MEMBER(mbox_min_index_size) 0,
@@ -364,7 +364,7 @@
 	MEMBER(pop3_enable_last) FALSE,
 	MEMBER(pop3_reuse_xuidl) FALSE,
 	MEMBER(pop3_lock_session) FALSE,
-	MEMBER(pop3_uidl_format) "",
+	MEMBER(pop3_uidl_format) "%08Xu%08Xv",
 	MEMBER(pop3_client_workarounds) "",
 	MEMBER(pop3_logout_format) "top=%t/%p, retr=%r/%b, del=%d/%m, size=%s",
 
--- dovecot-1.0.rc7/dovecot-example.conf.default-settings	2006-08-10 23:21:05.000000000 +0200
+++ dovecot-1.0.rc7/dovecot-example.conf	2006-09-22 11:32:56.000000000 +0200
@@ -7,16 +7,13 @@
 # value inside quotes, eg.: key = "# char and trailing whitespace  "
 
 # Default values are shown for each setting, it's not required to uncomment
-# any of the lines. Exception to this are paths, they're just examples with
-# the real defaults being based on configure options. The paths listed here
-# are for configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var
-# --with-ssldir=/etc/ssl
+# any of the lines.
 
 # Base directory where to store runtime data.
 #base_dir = /var/run/dovecot/
 
 # Protocols we want to be serving: imap imaps pop3 pop3s
-#protocols = imap imaps
+#protocols = imap imaps pop3 pop3s
 
 # IP or host address where to listen in for connections. It's not currently
 # possible to specify multiple addresses. "*" listens in all IPv4 interfaces.
@@ -35,13 +32,13 @@
 #     listen = *:10100
 #     ..
 #   }
-#listen = *
+#listen = [::]
 
 # Disable LOGIN command and all other plaintext authentications unless
 # SSL/TLS is used (LOGINDISABLED capability). Note that if the remote IP
 # matches the local IP (ie. you're connecting from the same computer), the
 # connection is considered secure and plaintext authentication is allowed.
-#disable_plaintext_auth = yes
+#disable_plaintext_auth = no
 
 # Should all IMAP and POP3 processes be killed when Dovecot master process
 # shuts down. Setting this to "no" means that Dovecot can be upgraded without
@@ -86,8 +83,8 @@
 # dropping root privileges, so keep the key file unreadable by anyone but
 # root. Included doc/mkcert.sh can be used to easily generate self-signed
 # certificate, just make sure to update the domains in dovecot-openssl.cnf
-#ssl_cert_file = /etc/ssl/certs/dovecot.pem
-#ssl_key_file = /etc/ssl/private/dovecot.pem
+#ssl_cert_file = /etc/pki/dovecot/certs/dovecot.pem
+#ssl_key_file = /etc/pki/dovecot/private/dovecot.pem
 
 # If key file is password protected, give the password here. Alternatively
 # give it when starting dovecot with -p parameter.
@@ -412,7 +409,7 @@
 # locking methods as well. Some operating systems don't allow using some of
 # them simultaneously.
 #mbox_read_locks = fcntl
-#mbox_write_locks = dotlock fcntl
+#mbox_write_locks = fcntl
 
 # Maximum time in seconds to wait for lock (all of them) before aborting.
 #mbox_lock_timeout = 300
@@ -576,7 +573,7 @@
   # installations. %08Xu%08Xv will be the new default, so use it for new
   # installations.
   #
-  #pop3_uidl_format = 
+  #pop3_uidl_format = %08Xu%08Xv
 
   # POP3 logout format string:
   #  %t - number of TOP commands
