diff -up dovecot-2.1.rc5/dovecot.service.in.privatetmp dovecot-2.1.rc5/dovecot.service.in
--- dovecot-2.1.rc5/dovecot.service.in.privatetmp	2012-02-07 15:15:54.977571774 +0100
+++ dovecot-2.1.rc5/dovecot.service.in	2012-02-07 15:15:54.977571775 +0100
@@ -6,6 +6,7 @@ After=local-fs.target network.target
 Type=simple
 ExecStart=@sbindir@/dovecot -F
 NonBlocking=yes
+PrivateTmp=true
 
 [Install]
 WantedBy=multi-user.target
